_posts/2018-12-10-snowplow-javascript-tracker-2.10.0-released/index.md

```diff
@@ -0,0 +1,162 @@
+discourse: true
+---
+
+We are pleased to announce a new release of the [Snowplow JavaScript Tracker][js-tracker]. [Version 2.10.0][2.10.0-tag] introduces global contexts, a set of powerful features for working with contexts. Also included are: more context features for link and form tracking, a new option to prevent exceptions from surfacing, and many other under-the-hood improvements, updates, and bugfixes.
+
+Read on below the fold for:
+
+1. [Global contexts](#global-contexts)
+2. [Error-handling](#error-supression)
+3. [Dynamic contexts for link and form tracking](#link-form-tracking)
+4. [Updates and bug fixes](#updates)
+5. [Upgrading](#upgrade)
+6. [Documentation and help](#doc)
+
+<!--more-->
+
+<h2 id="global-contexts">1. Global contexts</h2>
+
+This release introduces a set of powerful tracker features for controlling when contexts are added to events. Traditionally, trackers have offered predefined contexts that add contexts to every event when enabled. We refer to these new contexts with expanded capabilities as global contexts because each global context is evaluated when an event is sent.
+
+Global context capabilities:
+
+- adding contexts to every event
+- evaluating contexts dynamically, we allow user-supplied callbacks that optionally accept an argument containing important information (schema URI, event payload, event type)
+- adding contexts based on an event's schema URI
+- adding contexts based on the result of a user-supplied callback
+
+The following methods are added to the tracker:
+1. [`addGlobalContexts`][add-global]
+2. [`removeGlobalContexts`][remove-global]
+3. [`clearGlobalContexts`][clear-global]
+
+For a taste of the functionality, here we can define that a context is only sent for events with the schema vendor `com.acme.marketing`. Notice that `addGlobalContexts` must be supplied with an array of global contexts.
+
+{% highlight javascript %}
+let user_context = {
+  schema: 'iglu:com.acme.contexts/user_context/jsonschema/1-0-0'
+  data: {
+    userid: 1234,
+    name: 'john doe'
+  }
+};
+
+// if a global context has a conditional component (in this case, a schema rule set)
+// our global context must be supplied as an array, where the first element is
+// the conditional part and the context is the second element
+let global_context = [
+  {accept: 'iglu:com.acme.marketing/\*/jsonschema/1-0-0'},
+  user_context
+]
+
+window.snowplow('addGlobalContexts', [global_context]);
+{% endhighlight %}
+
+Full documentation can be found [here][global-contexts-docs].
+
+<h2 id="error-supression">2. Error-handling</h2>
+
+With this version, by default all exceptions generated by the tracker will not surface.
+
+You can disable this by setting the following value:
+
+{% highlight javascript %}
+window.snowplow('addGlobalContexts', [global_context]);
+{% endhighlight %}
+
+<h2 id="link-form-tracking">3. Dynamic contexts for link and form tracking</h2>
+
```

Comment:
This section doesn't really make sense to me. When you say:

> With this version, by default all exceptions generated by the tracker will not surface.

What exceptions are you talking about? It looks from:

```js
window.snowplow('addGlobalContexts', [global_context]);
```

That is is related to exceptions caused when you attempt to add a global context to an event? Can we clarify please?